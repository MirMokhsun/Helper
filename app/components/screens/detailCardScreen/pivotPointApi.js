export const getNamberLength = (number) => {
    let result;
    if (!number) {
        result = 0;
    } else {
        result = number.toString().length;
    }
    return result;
}

export const getStringFromNumberWithLength = (number, length) => {
    let result;
    if (!number && !length) {
        result = '';
    } else {
        const strinig = number.toString();
        result = strinig.substring(0, length)
    }
    return result;
}