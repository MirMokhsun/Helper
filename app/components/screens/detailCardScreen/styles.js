import { StyleSheet } from 'react-native';
import Utils from '../../../api/utils/utils';

export const styles = StyleSheet.create({
    container: {
        backgroundColor: '#F5F5F7',
        flex: 1,
    },
    indicatorContainer: {
        flex: 1,
        alignItems: 'center',
    },
    chartContainer: {
        flex: 1,
        marginBottom: Utils.isIphoneX ? 30 : 0,
    },
    containerTitle: {
        flexDirection: 'row',
        justifyContent: 'center',
        marginLeft: 34,
    },
    titleStyle: {
        fontFamily: 'Roboto-Regular',
        fontWeight: 'bold',
        lineHeight: 28,
        fontSize: 22,
        color: '#000756',
    },
    starWrapper: {
        marginLeft: 10,
        width: 24,
    },
    containerIndicator: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        marginVertical: 15,
        marginLeft: 18,
    },
    idicatortextStyle: {
        fontFamily: 'Roboto-Light',
        lineHeight: 18,
        fontSize: 14,
        marginRight: 10,
    },
});