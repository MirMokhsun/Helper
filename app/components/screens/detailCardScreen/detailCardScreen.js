import React, { Component } from 'react';
import { View, TouchableOpacity, Text, Platform, I18nManager } from 'react-native';
import { styles } from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setNullPivotPointCardItemRecomendation } from './redux/cardItemAction';
import StarSVG from '../../commonComponents/svg/starSVG';
import FullStar from '../../commonComponents/svg/FullStar';
import RightArrow from '../../commonComponents/svg/RightArrow';
import UpArrow from '../../commonComponents/svg/UpArrow';
import DownArrow from '../../commonComponents/svg/DownArrow';
import { AskSpredBidLine } from './detailCardScreenElements/askSpredBidLine/askSpredBidLine';
import { localization } from '../../../common/localization';
import { PivotPointAndChartsLine } from './detailCardScreenElements/pivotPointAndChartsLine/pivotPointAndChartsLine';
import TimeFrameCharts from './detailCardScreenElements/timeFrameCharts/timeFrameCharts';
import Chart from './detailCardScreenElements/chart/chart';
import { IndicatorApi } from '../../../api/indicatorApi/indicatorApi';
import { returnObjectLength } from '../../../api/funcWithObject/funcWithObject';
import { setZoomCharts } from './detailCardScreenElements/chart/redux/zoomChartsAction';

export class DetailCardScreen extends Component {

    static navigationOptions = ({ navigation }) => {
        return {
            headerTintColor: '#000756',
            headerBackTitleVisible: true,
            title: navigation.state.params ? navigation.state.params.title ? navigation.state.params.title : '' : '',
            headerTitleStyle: {
                color: "#000756",
                alignSelf: 'center',
                width: '100%',
                height: '100%',
                textAlign: "center",
                fontSize: 16,
                lineHeight: 40,
                ...Platform.select({
                    ios: {
                        paddingRight: 0,
                    },
                    android: {
                        paddingRight: I18nManager.isRTL ? 0 : 30,
                        paddingLeft: I18nManager.isRTL ? 30 : 0,
                    },
                }),
                fontFamily: 'Roboto-Regular',
                fontWeight: 'normal',
                backgroundColor: "#F5F5F7",
            },
            headerBackTitle: null,
            headerStyle: {
                backgroundColor: "#F5F5F7",
                paddingTop: Platform.OS === 'ios' ? 20 : 10,
                borderBottomColor: "transparent",
                shadowOpacity: 0,
                elevation: 0,
            },
            headerRight: (<View />),
        }
    };

    constructor(props) {
        super(props)
        const { navigation } = this.props;
        const { state: { params: { isCardFavorite: isCardFavoriteProps } } } = navigation;
        this.IndicatorApi = new IndicatorApi();
        this.state = {
            colorIndicator: '#9FA0B2',
            arrow: <RightArrow />,
            favoriteSvg: isCardFavoriteProps ? <FullStar /> : <StarSVG />,
            isCardFavorite: isCardFavoriteProps,
            isCandels: false,
        }
        this.svgWithBool = { favoriteSvg: isCardFavoriteProps ? <FullStar /> : <StarSVG />, isCardFavorite: isCardFavoriteProps };
    }

    componentDidMount = () => {
        const { recomendation } = this.state;
        this.setColor(recomendation);
    }

    componentWillReceiveProps(nextProps) {
        const { recomendation } = nextProps;
        this.setColor(recomendation.Summary);
    }

    componentWillUnmount() {
        const { setNullPivotPointCardItemRecomendation } = this.props;
        setNullPivotPointCardItemRecomendation();
    }

    choseCardFaforite = () => {
        const { isCardFavorite: isCardFavoriteState } = this.state;
        const { navigation: { state: { params: { choseCardFaforite: choseCardFaforiteProps } } } } = this.props;
        if (isCardFavoriteState) {
            this.ifCardFavoriteStateTrueFunction();
        } else {
            this.svgWithBool.favoriteSvg = <FullStar />;
            this.svgWithBool.isCardFavorite = true;
        }
        this.setState(this.svgWithBool);
        choseCardFaforiteProps();
    }

    ifCardFavoriteStateTrueFunction = () => {
        const { navigation: { state: { params: { getQuotationMap } } } } = this.props;
        const quotationMap = getQuotationMap();
        const favoritesObjLenght = returnObjectLength(quotationMap.favoritesObj);
        if (favoritesObjLenght > 1) {
            this.svgWithBool.favoriteSvg = <StarSVG />;
            this.svgWithBool.isCardFavorite = false;
        }
    }

    setColor = (recomendation) => {
        let result;
        switch (recomendation) {
            case 'Buy':
                result = { colorIndicator: '#1459D2', arrow: <UpArrow /> };
                break;
            case 'Sell':
                result = { colorIndicator: '#F86F6F', arrow: <DownArrow /> };
                break;
            case 'Neutral':
                result = { colorIndicator: '#9FA0B2', arrow: <RightArrow /> };
                break;
            default:
                /* code */
                break;
        }
        this.setState(result);
    }

    changeCharts = () => {
        const { isCandels } = this.state;
        this.setState({ isCandels: !isCandels });
    }

    render() {
        const { language, cardItem: { FullName, Ask, Bid, ID, Name }, cleanDatesArr, pivotPoint, recomendation, setZoomCharts } = this.props;
        const { favoriteSvg, arrow, colorIndicator, isCandels } = this.state;
        const local = localization[language];
        return (
            <View style={styles.container}>
                <View style={styles.indicatorContainer}>
                    <View style={styles.containerTitle}>
                        <Text style={styles.titleStyle}>{FullName}</Text>
                        <TouchableOpacity onPress={this.choseCardFaforite} style={styles.starWrapper} testID={`DetailCard${FullName}`} accessibilityLabel={`DetailCard${FullName}`}>
                            {favoriteSvg}
                        </TouchableOpacity>
                    </View>
                    <View style={styles.containerIndicator}>
                        <Text style={[styles.idicatortextStyle, { color: colorIndicator }]}>{local[recomendation.Summary]}</Text>
                        {arrow}
                    </View>
                    <AskSpredBidLine ask={Ask} bid={Bid} language={language} />
                    <PivotPointAndChartsLine changeCharts={this.changeCharts} pivotPoint={pivotPoint.Pivot} />
                    <TimeFrameCharts {...{ ID, local, cleanDatesArr, Name, chengeTimeFrame: this.chengeTimeFrame, setZoomCharts }} />
                </View>
                <View style={styles.chartContainer}>
                    <Chart isCandels={isCandels} />
                </View>
            </View >
        )
    }
}

const mapStateToProps = state => ({
    language: state.currentLanguage.language,
    cardItem: state.cardItem.item,
    pivotPoint: state.cardItem.pivotPoint,
    recomendation: state.cardItem.recomendation,
    cleanDatesArr: state.candlesFunc.cleanDatesArr,
});

const mapDispatchToProps = dispatch => bindActionCreators({
    setNullPivotPointCardItemRecomendation,
    setZoomCharts,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(DetailCardScreen);