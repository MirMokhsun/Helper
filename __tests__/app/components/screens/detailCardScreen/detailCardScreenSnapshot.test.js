import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { DetailCardScreen } from '../../../../../app/components/screens/detailCardScreen/detailCardScreen';

const mock = {
    FullName: 'sad',
    Ask: 0.0545,
    Bid: 0.445
}

let navigation = {
    "state": {
        "params": {

        }
    },
    setParams: () => { }
}

it('create snapshot DetailCardScreen', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<DetailCardScreen
        setCardItem={() => { }}
        setRegistrationPhoneCode={() => { }}
        cardItem={mock}
        language={'ru'}
        navigation={navigation}
        recomendation={{Summary:''}}
        pivotPoint={{Pivot:'123456'}}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});
