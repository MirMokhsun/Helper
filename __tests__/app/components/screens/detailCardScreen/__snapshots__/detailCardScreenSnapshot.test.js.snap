// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot DetailCardScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F5F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 34,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000756",
            "fontFamily": "Roboto-Regular",
            "fontSize": 22,
            "fontWeight": "bold",
            "lineHeight": 28,
          }
        }
      >
        sad
      </Text>
      <TouchableOpacity
        accessibilityLabel="DetailCardsad"
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "marginLeft": 10,
            "width": 24,
          }
        }
        testID="DetailCardsad"
      >
        <StarSVG />
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 18,
          "marginVertical": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Roboto-Light",
              "fontSize": 14,
              "lineHeight": 18,
              "marginRight": 10,
            },
            Object {
              "color": "#9FA0B2",
            },
          ]
        }
      />
      <RightArrow />
    </View>
    <AskSpredBidLine
      ask={0.0545}
      bid={0.445}
      language="ru"
    />
    <PivotPointAndChartsLine
      changeCharts={[Function]}
      pivotPoint="123456"
    />
    <TimeFrameCharts
      local={
        Object {
          "10 characters minimum": "Минимальное количество 10 символов",
          "A letter to confirm your data...": "Письмо для подтверждения Ваших данных и завершения процесса регистрации отправлено на указаный e-mail",
          "Already have an account?": "Уже есть аккаунт ?",
          "AnotherProblem": "Обнаружена непредвиденная ошибка",
          "Are you sure you want to log out?": "Вы уверены, что хотите выйти ?",
          "Authorization": "Авторизация",
          "Buy": "Покупать",
          "Cancel": "Отмена",
          "Catalog": "Каталог",
          "ConnectionError": "Не удалось войти из-за проблем с соединением. Проверьте подключение к Интернету",
          "ContactBroker": "Связь с брокером",
          "Country": "Страна",
          "Day": "день",
          "Email-entry error": "Ошибка ввода e-mail",
          "EmailIsExist": "Пользователь с таким e-mail уже существует. Найдите письмо с паролем на вашей почте или зарегистрируйтесь под новым e-mail адресом.",
          "EmailOrPassWrong": "Некорректный адрес электронной почты или пароль",
          "EmailWrong": "Некорректный адрес электронной почты",
          "Enter your country": "Укажите страну",
          "Favorites": "Избранное",
          "Forgot password?": "Забыли пароль?",
          "Glossary": "Словарь",
          "Help": "Помощь",
          "Home": "Главная",
          "Hour": "час",
          "I have read and agree with": "Заполняя форму регистрации, я принимаю",
          "Indicators": "Показатели",
          "Instructions for password recovery...": "Инструкция

 по восстановлению пароля отправлена на почту пользователя",
          "Instruments": "Инструменты",
          "Invalid email address or password": "Неверный адрес электронной почты или пароль",
          "Invalid email address or server error": "Неверный адрес электронной почты или ошибка сервера",
          "InvalidParameters": "Регистрация в данный момент недоступна. Повторите попытку позже",
          "LanguagePreference": "Выбор языка",
          "Languages": Object {
            "Arabian": "Арабский",
            "English": "Английский",
            "Russian": "Русский",
            "Spanish": "Испанский",
          },
          "Last tool cannot be removed from favorites": "Последний инструмент невозможно удалить из списка избранных ",
          "LastUpdate": "Последнее обновление",
          "Leave a comment": "Оставить комментарий",
          "Liked the application": "Понравилось приложение?
Оцените его!",
          "LogInToTheAccount": "Войти в аккаунт",
          "ManagerWillContact": "Менеджер скоро с Вами свяжется",
          "More news": "Ещё новости",
          "MovingAverages": "Скользящие средние",
          "Neutral": "Нейтрально",
          "News": "Новости",
          "No": "Нет",
          "OK": "ОК",
          "Other": "Прочее",
          "OurApps": "Торговое приложение",
          "OurWebsite": "Наш сайт",
          "Password failed": "Ошибка ввода пароля",
          "Password recovery": "Восстановление пароля",
          "PasswordRecoverySuccess": "Инструкция по восстановлению пароля отправлена на почту пользователя",
          "Please specify the email you used when registering": "Пожалуйста, укажите e-mail, который Вы использовали при регистрации",
          "Quote": "Котировка",
          "RegSuccess": "Регистрация прошла успешно",
          "Registration": "Регистрация",
          "Remind me later": "Напомнить позже",
          "Save": "Сохранить",
          "Sell": "Продавать",
          "ServerUnreachable": "Ошибка при подключении к серверу",
          "Session": "Сессия",
          "Settings": "Настройки",
          "Something went wrong": "что - то пошло не так",
          "Specification": "Детализация",
          "Thank you for joining us!": "Спасибо,

 что присоеденились к нам!",
          "Time": "Время",
          "To get full access to the instruments...": "Чтобы получить полный доступ к инструментам, Вам необходимо войти в систему",
          "Week": "нед",
          "Yes": "Да",
          "YouLoggedAs": "Вы вошли как ",
          "cellItem": Object {
            "opencellItem": Object {
              "closeTitle": "Закрытие:",
              "diferenceTitle": "Разница:",
              "maxTitle": "Макс.:",
              "minTitle": "Мин.:",
              "openTitle": "Открытие:",
              "percentdifereceTitle": "Разница%:",
            },
            "titleOfBuy": "Покупка",
            "titleOfSell": "Продажа",
          },
          "confirmPasswordHints": Object {
            "emptyName": "Поле обязательно для заполнения",
            "emptyNameSO": "Поле обязательно для заполнения",
            "length": "Пароль должен содержать не менее 5 символов",
            "lengthSO": "Пароль должен содержать не менее 5 символов",
            "minOneDigit": "В пароле должны присутствовать буквы и цифры",
            "minOneDigitSO": "В пароле должны присутствовать буквы и цифры",
            "minOneLeter": "В пароле должны присутствовать буквы и цифры",
            "minOneLeterSO": "В пароле должны присутствовать буквы и цифры",
            "onlyLatinAndDigits": "5-20 знаков, английские буквы и цифры",
            "onlyLatinAndDigitsSO": "5-20 знаков, английские буквы и цифры",
          },
          "confirmPasswordMatch": "Пароли должны совпадать",
          "countries": Object {
            "London": "Лондон",
            "Moscow": "Москва",
            "New York": "Нью-Йорк",
            "Sydney": "Сидней",
            "Tokyo": "Токио",
          },
          "createAccount": "Cоздать аккаунт",
          "emailExists": "Такой адрес уже существует",
          "emailHints": Object {
            "emptyName": "Поле обязательно для заполнения",
            "emptyNameSO": "Поле обязательно для заполнения",
            "incorrect": "Введите корректный формат эл. почты",
            "incorrectSO": "Введите корректный формат эл. почты",
            "symbols": "Допустимы символы \\".\\", \\"_\\", \\"-\\", \\"+\\", \\"@\\"",
            "symbolsSO": "Допустимы символы \\".\\", \\"_\\", \\"-\\", \\"@\\"",
          },
          "enter": "Вход",
          "error!": "Ошибка!",
          "errorMessage": Object {
            "email": "Неверный email",
            "lastName": "Неверная фамилия",
            "name": "Неверное имя",
            "phoneNumber": "Неверно указан номер телефона",
          },
          "errorTypes": Object {
            "3": "Слишком простой пароль",
            "6": "Пользователь с этим e-mail уже зарегистрирован.",
          },
          "errors": Object {
            "errorEmail": "Ошибка ввода e-mail",
            "errorFirstName": "Ошибка ввода имени",
            "errorLastName": "Ошибка ввода фамилии",
          },
          "exit": "Выход",
          "helpSearch": "Введите слова поиска",
          "introduced": "Ознакомлен",
          "lastNameHints": Object {
            "emptyName": "Поле обязательно для заполнения",
            "emptyNameSO": "Поле обязательно для заполнения",
            "incorrect": "Введите корректные данные",
            "incorrectSO": "Введите корректные данные",
          },
          "login": "Войти",
          "min": "мин",
          "nameHints": Object {
            "emptyName": "Поле обязательно для заполнения",
            "emptyNameSO": "Поле обязательно для заполнения",
            "incorrect": "Введите корректные данные",
            "incorrectSO": "Введите корректные данные",
          },
          "noMessage": "Уведомлений нет",
          "notifications": "Уведомления",
          "passwordHints": Object {
            "emptyName": "Поле обязательно для заполнения",
            "emptyNameSO": "Поле обязательно для заполнения",
            "length": "Пароль должен содержать не менее 5 символов",
            "lengthSO": "Пароль должен содержать не менее 5 символов",
            "minOneDigit": "В пароле должны присутствовать буквы и цифры",
            "minOneDigitSO": "В пароле должны присутствовать буквы и цифры",
            "minOneLeter": "В пароле должны присутствовать буквы и цифры",
            "minOneLeterSO": "В пароле должны присутствовать буквы и цифры",
            "onlyLatinAndDigits": "5-20 знаков, английские буквы и цифры",
            "onlyLatinAndDigitsSO": "5-20 знаков, английские буквы и цифры",
          },
          "phoneNumberHints": Object {
            "emptyName": "Поле обязательно для заполнения",
            "emptyNameSO": "Поле обязательно для заполнения",
            "incorrect": "Введите правильный номер телефона",
            "incorrectSO": "Введите правильный номер телефона",
            "onlyNumbers": "Введите только цифры",
            "onlyNumbersSO": "Введите только цифры",
          },
          "placeHolder": Object {
            "confirmPassword": "Подтверждение пароля",
            "email": "Email",
            "lastName": "Фамилия",
            "name": "Имя",
            "password": "Пароль",
            "phoneNumber": "Телефон",
          },
          "recvOffers": "Получить предложение",
          "regPopUp": "Ваш пароль отправлен на указанную почту",
          "sendBtnText": "Отправить",
          "signup": "Зарегистрируйтесь",
          "signup2": "Зарегистрироваться",
          "statusCodes": Object {
            "500": "Регистрация недоступна,повторите попытку.",
          },
          "stillDontHaveAccaunt": "Ещё нет учётной записи?",
          "tabbar": Object {
            "commodity": "Товары",
            "currency": "Валюты",
            "home": "Главная",
            "index": "Индексы",
            "stock": "Акции",
          },
          "thanks": "Спасибо!",
          "turnBtn": "Свернуть все",
          "user agreement": "условия обработки персональных данных.",
          "user agreement2": "Пользовательское соглашение",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": 30,
      }
    }
  >
    <Memo(Connect(Chart))
      isCandels={false}
    />
  </View>
</View>
`;
